File: Golang/Subarray Sums Divisible by K.go
func subarraysDivByK(nums []int, k int) int {
    var ans int = 0
    var sum int = 0
    Map := make(map[int]int)
    Map[0] = 1
    for i := 0; i < len(nums); i++ {
        sum = (((sum + nums[i]) % k) + k) % k
        ans += Map[sum]
        Map[sum]++
    }
    return ans
}

File: Python3/Subarray Sums Divisible by K.py
class Solution:
    def subarraysDivByK(self, nums: List[int], k: int) -> int:
        result, running_sum = 0,0
        mod_array = [1] + [0]*  k
        for i in range(len(nums)):
            running_sum += nums[i]
            result += mod_array[running_sum % k]
            mod_array[running_sum % k]+=1
        return result